src/SqlBindingUtilities.cs

```diff
@@ -135,30 +136,11 @@ public static SqlCommand BuildCommand(SqlAttribute attribute, SqlConnection conn
         /// <summary>
         /// Returns a dictionary where each key is a column name and each value is the SQL row's value for that column
         /// </summary>
-        /// <param name="reader">
-        /// Used to determine the columns of the table as well as the next SQL row to process
-        /// </param>
-        /// <param name="cols">
-        /// Filled with the columns of the table if empty, otherwise assumed to be populated
-        /// with their names already (used for cacheing so we don't retrieve the column names every time)
-        /// </param>
+        /// <param name="reader">Used to determine the columns of the table as well as the next SQL row to process</param>
         /// <returns>The built dictionary</returns>
-        public static Dictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader, List<string> cols)
+        public static IReadOnlyDictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader)
```

Comment:
Please make generic "cleanup" changes like this and the public -> internal sealed in a separate PR. They don't seem to be related to the feature and should be reviewed separately.